lib/page-object/loads_platform.rb

```diff
@@ -15,12 +15,31 @@ module LoadsPlatform
     # @returns [PageObject] 
     #
     def load_platform(browser, adapters)
-      adapters.each_value { |adapter|
-        return adapter.create_page_object(browser) if adapter.is_for?(browser)
+      adapter_for(browser,adapters).create_page_object(browser)
+    end
+
+    def browser_for root,adapters
+      adapter_for(root,adapters).browser_for(root)
+    end
+
+    def adapter_for element_or_browser, adapters
```

Comment:
extracted platform selection and corresponding error message into its own method for use with other methods